(docs) Add convention about installer arguments

AdmiringWorm opened the pull request:
## Description

This updates the COMMITTERS contribution documentation to include
information about what silent arguments that should be used with each of
the technologies that are commonly used.

## Motivation and Context

To try to standardize the conventions we use in this repository, or are moving towards using.

## How Has this Been Tested?

N/A

## Screenshot (if appropriate, usually isn't needed):

## Types of changes

~- [ ] Bug fix (non-breaking change which fixes an issue)~
~- [ ] New feature (non-breaking change which adds functionality)~
~- [ ] Breaking change (fix or feature that would cause existing functionality to change)~
~- [ ] Migrated package (a package has been migrated from another repository)~
- [x] Documentation

## Checklist:

~- [ ] My code follows the code style of this repository.~
~- [ ] My change requires a change to documentation (this usually means the notes in the description of a package).~
- [x] I have updated the documentation accordingly (this usually means the notes in the description of a package).
~- [ ] I have updated the package description and it is less than 4000 characters.~
~- [ ] All files are up to date with the latest [Contributing Guidelines](https://github.com/chocolatey-community/chocolatey-packages/blob/master/CONTRIBUTING.md)~
~- [ ] The added/modified package passed install/uninstall in the chocolatey test environment.~
~- [ ] The changes only affect a single package (not including meta package).~

pauby commented on COMMITTERS.md:
```diff
@@ -50,3 +50,31 @@ Before setting the `Version` parameter to the returned object in `au_GetLatest`,
+- **MSI**: Arguments for this technology should use ``/qn /norestart /l*v `"$($env:TEMP)\$($env:chocolateyPakcageName).$(env:chocolateyPackageVersion).MsiInstall.log`"``
+  - If the technology used to create the MSI is CMake, or WiX the additional argument `ALLUSERS=1` should also be used. `lessmsi` can be used to determine if the property `ALLUSERS` is available or not.
+- **Inno Setup**: Arguments used for this technology should use ``/VERYSILENT /NORESTART /SUPPRESSMSGBOXES /SP- /LOG=`"$($env:TEMP)\$($env:chocolateyPackageName).$($env:chocolateyPackageVersion).InnoInstall.log`"``
+  - If additionally the installer uses version 6.x of Inno Setup, the argument `/ALLUSERS` should also be used. This argument is specified when running `installer_name.exe /HELP`
```
Is this just v6.x or v6.x and later?

pauby replied:
Can we clarify what is meant by :

> This argument is specified when running `installer_name.exe /HELP`

It's not clear to me.

AdmiringWorm replied:
Currently, Inno Setup is on v6.2.2, while I can assume that they will continue supporting this argument in v7.x, I can't say for sure.

I'll probably update this list with links to the documentation of each technology that I know where the documentation is located.

pauby commented on COMMITTERS.md:
```diff
@@ -50,3 +50,31 @@ Before setting the `Version` parameter to the returned object in `au_GetLatest`,
+  - If the technology used to create the MSI is CMake, or WiX the additional argument `ALLUSERS=1` should also be used. `lessmsi` can be used to determine if the property `ALLUSERS` is available or not.
+- **Inno Setup**: Arguments used for this technology should use ``/VERYSILENT /NORESTART /SUPPRESSMSGBOXES /SP- /LOG=`"$($env:TEMP)\$($env:chocolateyPackageName).$($env:chocolateyPackageVersion).InnoInstall.log`"``
+  - If additionally the installer uses version 6.x of Inno Setup, the argument `/ALLUSERS` should also be used. This argument is specified when running `installer_name.exe /HELP`
+- **NSIS**: Arguments used for this technology should be `/S`. Additionally, investigation is necessary to see if the authors of the software has implemented a way to specify a log file as well.
```
```suggestion
- **NSIS**: Arguments used for this technology should be `/S`. Additionally, investigation is necessary to see if the authors of the software have implemented a way to specify a log file as well.
```

pauby commented on COMMITTERS.md:
```diff
@@ -50,3 +50,31 @@ Before setting the `Version` parameter to the returned object in `au_GetLatest`,
+- **Inno Setup**: Arguments used for this technology should use ``/VERYSILENT /NORESTART /SUPPRESSMSGBOXES /SP- /LOG=`"$($env:TEMP)\$($env:chocolateyPackageName).$($env:chocolateyPackageVersion).InnoInstall.log`"``
+  - If additionally the installer uses version 6.x of Inno Setup, the argument `/ALLUSERS` should also be used. This argument is specified when running `installer_name.exe /HELP`
+- **NSIS**: Arguments used for this technology should be `/S`. Additionally, investigation is necessary to see if the authors of the software has implemented a way to specify a log file as well.
+- **InstallShield**: The arguments for this technology can vary based on whether it has an embedded MSI or not. For normal non-MSI builds, the argument `/s` should be used, but if it uses an embedded MSI the arguments `/s /v"/qn"` should be used.
```
```suggestion
- **InstallShield**: The arguments for this technology can vary based on whether it has an embedded MSI or not. For normal non-MSI installers, the argument `/s` should be used, but if it uses an embedded MSI installer, the arguments `/s /v"/qn"` should be used.
```

pauby commented on COMMITTERS.md:
```diff
@@ -50,3 +50,31 @@ Before setting the `Version` parameter to the returned object in `au_GetLatest`,
+- **NSIS**: Arguments used for this technology should be `/S`. Additionally, investigation is necessary to see if the authors of the software has implemented a way to specify a log file as well.
+- **InstallShield**: The arguments for this technology can vary based on whether it has an embedded MSI or not. For normal non-MSI builds, the argument `/s` should be used, but if it uses an embedded MSI the arguments `/s /v"/qn"` should be used.
+- **Wise InstallMaster**: Arguments used for this technology should use `/s`.
+- **Ghost** or **Squirrel**: Arguments used for these technologies should use `-s`. These technologies are best avoided, as they can only be installed for the current user. Use alternative installers if possible (*must be provided officially*).
```
```suggestion
- **Ghost** or **Squirrel**: Arguments used for these technologies should use `-s`. These technologies are best avoided, as they can only be installed for the current user. Use alternative official installers if possible.
```

pauby reviewed (dismissed):
This is a really useful addition @AdmiringWorm. Thanks for putting it together.

Added a few questions and tweaks.

AdmiringWorm commented on COMMITTERS.md:
```diff
@@ -50,3 +50,31 @@ Before setting the `Version` parameter to the returned object in `au_GetLatest`,
+To determine installers, and the possible support of arguments there are three utilities available on the Chocolatey Community Repository that may be used, and it is expected that anyone working on this repository have these installed.
+
+1. The easiest utility to use to determine an installer, is the [Universal Silent Switch Finder](https://community.chocolatey.org/packages/ussf). When it determines the installer, it also suggests what the arguments to use to make the installation silent. Verify these arguments with the below information of what we want to be used.
+2. A more advanced utility to use is the [TrID - File Identifier](https://community.chocolatey.org/packages/trid). This is a more up-to-date utility that can handle different file types, and not just installers. This unfortunately do not typically handle NSIS installers, and will not give information about possible silent arguments.
```
The reason I went with `typically` here is because there are some instances (probably some versions of NSIS) that TRiD can detect.

AdmiringWorm commented on COMMITTERS.md:
```diff
@@ -50,3 +50,31 @@ Before setting the `Version` parameter to the returned object in `au_GetLatest`,
+3. The third utility to use is only available in some cases for MSI installers. The utility [LessMSI](https://community.chocolatey.org/packages/lessmsi) can be used for MSI installers to find out what properties are available in the installer. Typically, these properties can be used as part of silent arguments. Usually we are only interested if the property `ALLUSERS` is available to be used here ...
+
+The following list shows what we expected each technology to use as their silent arguments.
+All arguments should follow the same casing as defined below. Arguments should also be specified in a single quote (`'`) if they do not use any variables, and use double quotes (`"`) if any arguments are used.
```
That is what I was intending to say yes. Although, when thinking about it.

I'll try and see how I can reword this.

AdmiringWorm commented:
@pauby I have rewritten just about the entire section, and changed the list of the installer technologies to have their own section with a bit more information. I also think I caught all the grammar errors you noted (if I didn't I'll blame LanguageTool).

Please let me know if something doesn't make any sense.

JPRuskin commented on COMMITTERS.md:
```diff
@@ -50,3 +50,162 @@ Before setting the `Version` parameter to the returned object in `au_GetLatest`,
+
+To determine installers and the possible support of arguments, there are three utilities available on the Chocolatey Community Repository that may be used, and it is expected that anyone working on this repository has these installed.
+
+1. The easiest utility to use to determine an installer is the [Universal Silent Switch Finder][]. When it determines the installer, it also suggests what the arguments to use to make the installation silent. Verify these arguments with the below information about what should be used in this repository.
```
Arguably either this or ending the sentence with an "are"?

JPRuskin reviewed (commented):
Looks great!

JPRuskin reviewed (approved):
I think this looks good.
